diff --git a/waypaper/changer.py b/waypaper/changer.py
--- a/waypaper/changer.py
+++ b/waypaper/changer.py
@@ -1,5 +1,6 @@
 """Module that runs the system processes to change the wallpaper"""
 
+import shlex
 import subprocess
 import time
 from typing import Optional
@@ -403,9 +404,8 @@
 
         # Run a post command:
         if cf.post_command and cf.use_post_command:
-            modified_image_path = str(image_path).replace(" ", "\\ ")
-            post_command = cf.post_command.replace("$wallpaper", modified_image_path)
-            post_command = post_command.replace("$monitor", monitor)
+            post_command = cf.post_command.replace("$monitor", shlex.quote(monitor))
+            post_command = post_command.replace("$wallpaper", shlex.quote(str(image_path)))
             subprocess.Popen(post_command, shell=True)
             print(f'Executed "{post_command}" post-command\n')
 
